rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,429 @@
+| breaks when interrupted with newlines | Yes | No | ? | No | |
+| Simplicity (Brainload) | Medium | Simple | Complex | More Complex | |
+
+### Refactoring note
+
+**Observing**: From a refactoring perspective, it might also be interesting to see how many conflicts the different formats would cause.
+
+nixpkgs comments:
+
+- `##` ~4k usages
+- `#` ~20k usages
+- `/*` ~6k usages
+- `/**` 160 usages 
+
+Choosing `/**` or subsets would cause minor conflicts within current nixpkgs. While this is NOT the main reason for the final decision, it MUST be considered.
+
+## Just free text as a content format
+
+While this allows the most freedom, it is usually considered the best option, not creating any restrictions.
+
+- [RFC72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md) defines commonMark as the official documentation format.
+This is why we decided to follow this convention. Writing plain text is still possible.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- Migration path for nixpkgs comments.
+- How to document the `arguments`. Should there be some markdown equivalent to `@param` i.e. `# Params`?
+    - Possible answer: Future RFC might need to clarify that.
+
+# Future work
+[Future]: #future-work
+
+## Migrate existing nixpkgs comments
+
+Reformatting existing doc-comments in nixpkgs.
+
+## nixpkgs Manual
+
+The current nixpkgs manual needs to be adopted to this change.
+
+## Native support in Nix
+
+- `NixOS/nix` should implement native support for doc-comments.
+
+> Note: We considered implementation details, but specifying those is out of scope for this rfc.
+
+Related POCs:
+
+- https://github.com/NixOS/nix/pull/9054
+- https://github.com/NixOS/nix/pull/1652
+- https://github.com/NixOS/nix/pull/5527
+- https://github.com/NixOS/nix/commit/8e252320f52426c734883709eb398d2161c3fe82
+
+## References
+
+### Other Conventions
+
+- [Rust](https://doc.rust-lang.org/stable/reference/comments.html#doc-comments)
+- [Python](https://peps.python.org/pep-0257/)
+- [JSDoc](https://jsdoc.app/)
+- [Go Doc Comments](https://go.dev/doc/comment)
+
+### Related tools
+
+- [Nixdoc](https://github.com/nix-community/nixdoc)
+- [Rustdoc](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html)
+- [Doxygen](https://www.doxygen.nl/)
+
+## Related discussions
+
+ - https://discourse.nixos.org/t/2023-04-13-documentation-team-meeting-notes-41/27264)
+
+- https://github.com/NixOS/nix/issues/3904
+- https://github.com/NixOS/nix/issues/228
+
+
+- https://github.com/NixOS/nix/pull/5527
+- https://github.com/NixOS/nix/pull/1652 
+
+[#1652](https://github.com/NixOS/nix/pull/1652) gets closest what this RFC imagines, implementation is almost done. It certainly has some limitations, but got stuck on politics and diverging opinions about markdown and syntax rules. 
```

Comment:
The tone of the last sentence is not very RFC-y IMO.

```suggestion
- [#1652](https://github.com/NixOS/nix/pull/1652) gets closest what this RFC imagines; the implementation is almost done, at the time of writing this RFC.
```

Reply:
```suggestion
[#1652](https://github.com/NixOS/nix/pull/1652) gets closest what this RFC imagines, implementation is almost done. It certainly has some limitations, but was not merged due to uncertainty and complexity. 
```